mahta/JS/JS data fetching/script.js

```diff
@@ -0,0 +1,174 @@
+fetch("https://jsonplaceholder.typicode.com/posts")
+  .then((response) => response.json())
+  .then((posts) => {
+    const firstTenPosts = posts.slice(0, 10);
+
+    const postsContainer = document.getElementById("posts-container");
+
+    firstTenPosts.forEach((post) => {
+      const postContainer = document.createElement("div");
+      postContainer.classList.add("post-container");
+
+      const title = document.createElement("h2");
+      title.classList.add("post-title");
+      title.textContent = post.title;
+
+      const body = document.createElement("p");
+      body.classList.add("post-body");
+      body.textContent = post.body;
+
+      const author = document.createElement("p");
+      author.classList.add("post-author");
+      fetch(`https://jsonplaceholder.typicode.com/users/${post.userId}`)
+        .then((response) => response.json())
+        .then((user) => {
+          // Set the author's name
+          author.textContent = `Author: ${user.name}`;
+        })
+        .catch((error) => {
+          console.error(error);
+          author.textContent = "Author: Unknown";
+        });
+
+      const updateButton = document.createElement("button");
+      updateButton.textContent = "Update";
+
+      updateButton.addEventListener("click", () => {
+        console.log(`Update post with ID ${post.id}`);
+
+        const updatedBody = prompt("Enter the updated body:", post.body);
+
+        if (updatedBody !== null) {
+          body.textContent = "Updating...";
+          fetch(`https://jsonplaceholder.typicode.com/posts/${post.id}`, {
+            method: "PATCH",
+            body: JSON.stringify({ body: updatedBody }),
```

Comment:
An example for updating posts (body):
[link](https://jsonplaceholder.typicode.com/guide/)

```js
body: JSON.stringify({
    id: 1,
    title: 'foo',
    body: 'bar',
    userId: 1,
  }),  
```